### PR TITLE
Removes redunant code and allows override query relation in descendan…

### PR DESCRIPTION
**Pull request by leandrogehlen**
| Q             | A
| ------------- | ---
| Is bugfix?    |  no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | 

Sometimes it's necessary to change default behavior for relation queries.

Example:

I have created a `MultiTenantRecord` where `ActiveRecord::find()` must be overrided, like:

```php

public static function find()
{
    $identity = Yii::$app->user->identity;
    if ($identity instanceof TenantInterface) {
        return parent::find()->where([static::tableName().'.tenant_id' => $identity->getTenantId()]);
    } else {
        throw new NotSupportedException("Identity does not implements TenantInteface");
    }
}
```
but this should be occurs only "master" record not for relation.
Then this allows customize query for relation

### CODE REVIEW

File: framework/db/BaseActiveRecord.php [hunk 1]
```diff
@@ -412,12 +406,24 @@ public function hasOne($class, $link)
...
+    }
+
+    /**
+     * Creates a query for `has-one` and `has-many` relation.
+     * @param string $class the class name of the related record
+     * @param array $link the primary-foreign key constraint
+     * @param boolean whether this query represents a relation to more than one record.
+     * @return ActiveQueryInterface the relational query object.
+     */
+    protected function newRelationQuery($class, $link, $multiple)
```
Comment (dynasource) [inline framework/db/BaseActiveRecord.php]: this should be called `createRelationQuery` and if we want to be consistent even `createLinkQuery`

**Review by dynasource — CHANGES_REQUESTED** [review]
this is useful for further customizing relations within ActiveRecords

File: framework/db/BaseActiveRecord.php (hunk 1 above)
Comment (leandrogehlen) [inline framework/db/BaseActiveRecord.php]: 👍 , @dynasource  i think better `createRelationQuery`

File: framework/db/BaseActiveRecord.php [hunk 2]
```diff
@@ -412,12 +406,24 @@ public function hasOne($class, $link)
...
     {
+        return $this->newRelationQuery($class, $link, true);
+    }
+
+    /**
+     * Creates a query for `has-one` and `has-many` relation.
+     * @param string $class the class name of the related record
+     * @param array $link the primary-foreign key constraint
+     * @param boolean whether this query represents a relation to more than one record.
+     * @return ActiveQueryInterface the relational query object.
```
Comment (SilverFire) [inline framework/db/BaseActiveRecord.php]: `@since 2.0.12`

**Review by SilverFire — CHANGES_REQUESTED** [review]
composer.lock should not be comminted

**Comment by leandrogehlen** [comment]
Sorry, i removed my fork accidentally, i will redo the changes in another PR

**Pull request closed without merging**